main.go

```diff
@@ -77,11 +77,12 @@ func sourceFromArg(arg string) (*source, error) {
 
 	// a GitHub or GitLab URL (even without the protocol):
 	var dnsErr *net.DNSError
+	var opErr *net.OpError
```

Comment:
if you are testing net.OpError

then you shouldn't test net.DNSError too

Here are some explanations:
https://stackoverflow.com/questions/69568234/how-do-i-work-out-what-type-of-error-my-error-is-in-go
https://haisum.github.io/2021/08/14/2021-golang-http-errors/

maybe url.Error could be enough for you